index.bs

```diff
@@ -318,6 +339,285 @@ NOTE: The size of the last layer can be determined by subtracting the sum of the
+
+```c
+aligned(8) class SampleTransform {
+    unsigned int(2) version = 0;
+    unsigned int(4) reserved;
+    unsigned int(2) bit_depth; // Signed 8, 16, 32 or 64-bit.
+    unsigned int(8) token_count;
+    for (i=0; i<token_count; i++) {
+        unsigned int(8) token;
+        if (token == 0) {
+            // Push the 'constant' value to the stack.
+            signed int(1<<(bit_depth+3)) constant;
+        } else if (token == 1) {
+            // Push the sample value from the 'input_image_item_index'th
+            // input image item to the stack. At most 'reference_count'.
+            unsigned int(8) input_image_item_index; // 1-based
+        } else {
+            if (token == 2 || token == 3 || token == 11 || token == 12) {
+                // Unary operator. Pop an operand from the stack.
+            } else {
+                // Binary operator. Pop a right operand
+                // then a left operand from the stack.
+            }
+            // Apply operator 'token' and push the result to the stack.
+        }
+    }
+}
+```
+
+<h5 id="sample-transform-semantics">Semantics</h5>
+
+<dfn noexport>version</dfn> shall be equal to 0.
+
+<dfn noexport>reserved</dfn> for future use.
+
+<dfn noexport>bit_depth</dfn> determines the precision (from 8 to 64 bits, see Table 1) of the signed integer temporary variable supporting the intermediate results of the operations. It also determines the precision of the stack elements and the field size of the <code>constant</code> fields. This intermediate precision shall be high enough so that all input sample values fit into that signed bit depth.
+
+<table class="data">
+  <caption style="caption-side:bottom">
+    Table 1 - Mapping from <code>bit_depth</code> to the intermediate bit depth.
+  </caption>
+  <thead>
+    <tr>
+      <th>Value of <code>bit_depth</code></th>
+      <th>Intermediate bit depth (sign bit inclusive) <code>num_bits</code></th>
+    </tr>
+  </thead>
+  <tbody>
+    <tr><td>0</td><td>8</td></tr>
+    <tr><td>1</td><td>16</td></tr>
+    <tr><td>2</td><td>32</td></tr>
+    <tr><td>3</td><td>64</td></tr>
+  </tbody>
+</table>
+
+The result of any computation under- or overflowing the intermediate bit depth is replaced by -2<sup><code>num_bits</code>-1</sup> and 2<sup><code>num_bits</code>-1</sup>-1, respectively. Encoder implementations should not create files leading to potential computation underflow or overflow. Decoder implementations shall check for computation underflow or overflow and adapt the results accordingly. Computations with operands of negative value use the two’s-complement representation.
```

Comment:
Nit: Change "operands of negative value" to "operands of negative values" (plural "values") or "negative operands".